docs/source/database_schema.rst

```diff
@@ -0,0 +1,60 @@
+Database Schema
+===============
+
+.. code-block:: sql
+
+
+    CREATE TABLE package_rules (
+        scan_id UUID NOT NULL,
+        rule_id UUID NOT NULL,
+        PRIMARY KEY (scan_id, rule_id),
+        FOREIGN KEY (scan_id) REFERENCES scans(scan_id),
+        FOREIGN KEY (rule_id) REFERENCES rules(id)
+    );
+
+    CREATE TABLE scans (
+        scan_id UUID PRIMARY KEY,
+        name VARCHAR,
+        version VARCHAR,
+        status VARCHAR,
+        score INT,
+        inspector_url VARCHAR,
+        queued_at TIMESTAMPTZ NULL DEFAULT CURRENT_TIMESTAMP,
+        queued_by VARCHAR,
+        pending_at TIMESTAMPTZ,
+        pending_by VARCHAR,
+        finished_at TIMESTAMPTZ,
+        finished_by VARCHAR,
+        reported_at TIMESTAMPTZ,
+        reported_by VARCHAR,
+        fail_reason VARCHAR,
+        commit_hash VARCHAR ,
+        UNIQUE (name, version)
+    );
+
+
+    CREATE UNIQUE INDEX idx_scans_name_version ON scans(name, version);
+    CREATE INDEX idx_scans_status ON scans(status) 
+        WHERE status IN ('QUEUED', 'PENDING');
+
+
+    CREATE TABLE download_urls (
+        id UUID PRIMARY KEY,
+        scan_id UUID NOT NULL,
+        url VARCHAR NOT NULL,
+        FOREIGN KEY (scan_id) REFERENCES scans(scan_id)
+    );
+
+
+    CREATE TABLE rules (
+        id UUID PRIMARY KEY,
+        name VARCHAR UNIQUE NOT NULL
+    );
+
```

Comment:
```suggestion
    CREATE EXTENSION IF NOT EXISTS pg_stat_statements WITH SCHEMA public;

    CREATE TYPE public.status AS ENUM (
        'QUEUED',
        'PENDING',
        'FINISHED',
        'FAILED'
    );

    CREATE TABLE public.download_urls (
        id uuid NOT NULL,
        scan_id uuid NOT NULL,
        url text NOT NULL
    );

    CREATE TABLE public.package_rules (
        scan_id uuid NOT NULL,
        rule_id uuid NOT NULL
    );

    CREATE TABLE public.rules (
        name text NOT NULL,
        id uuid NOT NULL
    );

    CREATE TABLE public.scans (
        scan_id uuid NOT NULL,
        name text NOT NULL,
        status public.status NOT NULL,
        score integer,
        version text NOT NULL,
        queued_at timestamp without time zone,
        pending_at timestamp without time zone,
        finished_at timestamp without time zone,
        reported_at timestamp without time zone,
        inspector_url text,
        reported_by text,
        queued_by text NOT NULL,
        pending_by text,
        finished_by text,
        commit_hash text,
        fail_reason text
    );

    ALTER TABLE ONLY public.download_urls
        ADD CONSTRAINT download_urls_pkey PRIMARY KEY (id);

    ALTER TABLE ONLY public.scans
        ADD CONSTRAINT name_version_unique UNIQUE (name, version);

    ALTER TABLE ONLY public.package_rules
        ADD CONSTRAINT package_rules_pkey PRIMARY KEY (scan_id, rule_id);

    ALTER TABLE ONLY public.scans
        ADD CONSTRAINT packages_pkey PRIMARY KEY (scan_id);

    ALTER TABLE ONLY public.rules
        ADD CONSTRAINT rules_name_key UNIQUE (name);

    ALTER TABLE ONLY public.rules
        ADD CONSTRAINT rules_pkey PRIMARY KEY (id);

    CREATE INDEX ix_download_urls_scan_id ON public.download_urls USING btree (scan_id);

    CREATE INDEX ix_scans_finished_at ON public.scans USING btree (finished_at);

    CREATE INDEX ix_scans_status ON public.scans USING btree (status) WHERE ((status = 'QUEUED'::public.status) OR (status = 'PENDING'::public.status));

    ALTER TABLE ONLY public.download_urls
        ADD CONSTRAINT download_urls_scan_id_fkey FOREIGN KEY (scan_id) REFERENCES public.scans(scan_id);

    ALTER TABLE ONLY public.package_rules
        ADD CONSTRAINT package_rules_rule_id_fkey FOREIGN KEY (rule_id) REFERENCES public.rules(id);

    ALTER TABLE ONLY public.package_rules
        ADD CONSTRAINT package_rules_scan_id_fkey FOREIGN KEY (scan_id) REFERENCES public.scans(scan_id);
```
add `pg_stat_statements` extension, `status` enum. also reorders tables and indexes so it is runnable, where previously it would fail due to the order